core/server/worker/src/main/java/alluxio/worker/block/meta/DefaultStorageDir.java

```diff
@@ -382,18 +382,23 @@ public long getReservedBytes() {
   }
 
   private void reclaimSpace(long size, boolean committed) {
-    Preconditions.checkState(mCapacityBytes >= mAvailableBytes.get() + size,
-        "Available bytes should always be less than total capacity bytes");
-    mAvailableBytes.addAndGet(size);
+    mAvailableBytes.getAndUpdate(oldAvailableBytes -> {
+      long newAvailableBytes = oldAvailableBytes + size;
+      Preconditions.checkState(mCapacityBytes >= newAvailableBytes,
+              "Available bytes should always be less than total capacity bytes");
+      return newAvailableBytes;
+    });
     if (committed) {
       mCommittedBytes.addAndGet(-size);
     }
   }
 
   private void reserveSpace(long size, boolean committed) {
-    Preconditions.checkState(size <= mAvailableBytes.get() + mReservedBytes.get(),
-        "Available bytes should always be non-negative");
-    mAvailableBytes.getAndSet(Math.max(0, mAvailableBytes.get() - size));
+    mAvailableBytes.getAndUpdate(oldAvailableBytes -> {
+      Preconditions.checkState(size <= oldAvailableBytes + getReservedBytes(),
+              "Available bytes should always be non-negative");
+      return oldAvailableBytes - size;
```

Comment:
If there is no available bytes, then require space less than reserved bytes, It will sucess and don't change  the value of available bytes. If afterwards reclaim space the value of available bytes will greater than real available bytes.
In this case, It is reasonable for available bytes to have a negative number.